### DISCUSSION TITLE
Making an atomFamily loadable in a safe way

### DISCUSSION THREAD

**Discussion by hemm1** [pmndrs/jotai#2733] [Q&A] [answered]
Hello! My team and me are just starting to migrate to Jotai from Recoil on all our applications. We really like the simplicity so far!

Today, we came across an issue: We have an `atomFamily`(in Jotai) which we sometimes want to use without suspending. We can use `loadable` for this, but we cannot figure out a good way to wrap an `atomFamily` with `loadable` without doing it directly in the react render code. We noticed that the documentation never seems to use `loadable` in the render code. Is there any downsides to doing that?

To be clear, we want to this: `const fuelPriceLoadable = useAtomValue(loadable(fuelPriceForType(fossilFuelType)));`

**Comment by dai-shi** [comment, marked as answer]
No, there're no downsides.

> documentation never seems to use loadable in the render code

Good point. We basically recommend using utils such as loadable outside render if possible. We should probably improve the doc about it.

In your case, it's not possible because `fossilFuelType` comes from props presumably.
Using loadable in render is supported through adding util-level caching.
For anyone interested, check out the code:
https://github.com/pmndrs/jotai/blob/9b835851a57b5e7333aa4c52a7f1833bea24b8e9/src/vanilla/utils/loadable.ts#L4-L6

**Reply by hemm1** [reply]
Thank you very much for you quick and helpful reply 🙏 😄
